ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiUserManagementService.scala

```diff
@@ -289,17 +307,30 @@ private[apiserver] final class ApiUserManagementService(
         )
       } yield (userId, rights, identityProviderId)
     ) { case (userId, rights, identityProviderId) =>
-      verifyUserIsNonExistentOrInIdp(identityProviderId, userId)
-        .flatMap { _ => userManagementStore.grantRights(id = userId, rights = rights) }
-        .flatMap(handleResult("grant user rights"))
-        .map(_.view.map(toProtoRight).toList)
-        .map(proto.GrantUserRightsResponse(_))
+      for {
+        _ <- verifyUserIsNonExistentOrInIdp(identityProviderId, userId)
```

Comment:
This is confusing: a user not-existing will result in a success from this method call.
Can you add a comment that this is nevertheless expected to result in an error from the user store layer?